subsys/bluetooth/mesh/heartbeat.c

```diff
@@ -100,8 +100,10 @@ static int heartbeat_send(const struct bt_mesh_send_cb *cb, void *cb_data)
 		.xmit = bt_mesh_net_transmit_get(),
 	};
 
-	/* Do nothing if heartbeat publication is not enabled */
-	if (pub.dst == BT_MESH_ADDR_UNASSIGNED) {
+	/* Do nothing if heartbeat publication is not enabled or the subnet is
+	 * removed.
+	 */
+	if (!tx.sub || pub.dst == BT_MESH_ADDR_UNASSIGNED) {
 		return 0U;
```

Comment:
I know you didn't touch this, but the `0U` return is odd since the return value is not unsigned, i.e. the `U` should probably be removed (can be a separate patch)

Reply:
True. I'll add a commit to this pr